aws-lc-rs/src/rsa.rs

```diff
@@ -130,17 +131,19 @@ impl RsaKeyPair {
     /// `error:KeyRejected` on error.
     pub fn from_der(input: &[u8]) -> Result<Self, KeyRejected> {
         unsafe {
-            let rsa = build_private_RSA(input)?;
-            Self::validate_rsa(&rsa.as_const())?;
-            Self::new(rsa)
+            let pkey = build_private_RSA(input)?;
+            Self::validate_rsa(&pkey)?;
+            Self::new(pkey)
         }
     }
     const MIN_RSA_BITS: u32 = 1024;
     const MAX_RSA_BITS: u32 = 2048;
 
-    unsafe fn validate_rsa(rsa: &ConstPointer<RSA>) -> Result<(), KeyRejected> {
-        let p = ConstPointer::new(RSA_get0_p(**rsa))?;
-        let q = ConstPointer::new(RSA_get0_q(**rsa))?;
+    unsafe fn validate_rsa(rsa: &LcPtr<*mut EVP_PKEY>) -> Result<(), KeyRejected> {
```

Comment:
```suggestion
    unsafe fn validate_rsa_pkey(rsa: &LcPtr<*mut EVP_PKEY>) -> Result<(), KeyRejected> {
```
Since we're validating the EVP_PKEY with an rsa type, instead of an `RSA` struct now.